### DISCUSSION TITLE
Immich android app sets a strange fileCreatedAt value on files it uploads

### DISCUSSION THREAD

**Discussion by oryjkov** [immich-app/immich#11044] [Q&A]
I have a photo named `1720461810927.jpg`. Filename is the date of export in ms and converts to `2024-07-08T18:03:30.927Z`.

Why is immich reporting the file to have `dateTimeOriginal` of `2024-07-08T18:03:51.000Z`? and not **`:31`** as it is in the original file? I'm guessing **`:51`** is the time of upload, but not really sure of that either. Any reference on what all these times are supposed to mean?

`exiftool` on the original file (the file that was uploaded to immich via the android app)
```
File Modification Date/Time     : 2024:07:08 20:03:31+02:00
File Access Date/Time           : 2024:07:08 20:03:31+02:00
File Inode Change Date/Time     : 2024:07:08 20:03:31+02:00
Modify Date                     : 2024:07:08 20:03:31
```

However on immich i get the following metadata - note that `fileModifiedAt` corresponds to the "correct" time and is **before** the time in fileCreatedAt.
```
  "fileCreatedAt": "2024-07-08T18:03:51.000Z",
  "fileModifiedAt": "2024-07-08T18:03:31.000Z",
  "localDateTime": "2024-07-08T18:03:51.000Z",
  "exifInfo": {
    "dateTimeOriginal": "2024-07-08T18:03:51.000Z",
    "modifyDate": "2024-07-08T20:03:31.000Z",
  }
```
Finally `exiftool` on immich asset file:
```
File Modification Date/Time     : 2024:07:08 20:03:31+02:00
File Access Date/Time           : 0000:00:00 00:00:00
File Inode Change Date/Time     : 2024:07:10 00:15:53+02:00
Modify Date                     : 2024:07:08 20:03:31
```

I'm trying to match up immich files with those in gphoto and this is throwing it off.

**Comment by bo0tzz** [comment]
Immich looks for these EXIF tags in order
https://github.com/immich-app/immich/blob/d68bd876c1a2d969b9749d0eb906ba7a810a54d7/server/src/services/metadata.service.ts#L40-L49
If none of them are set, it falls back to the time the file was uploaded to Immich.
You should make sure your files have the proper EXIF data. If you're importing a google takeout, you might want to try https://github.com/simulot/immich-go.

**Reply by oryjkov** [reply]
Thanks for the code link - this helps. The file is created by an app that just did not set the DateTimeOriginal.

> If none of them are set, it falls back to the time the file was uploaded to Immich.

Is that really the case, or it takes the time from the upload API parameter [fileModifiedAt](https://immich.app/docs/api/upload-asset)?

**Reply by bo0tzz** [reply]
That could be the case too, yeah.
